src/components/side_panel/components/collapse/collapse.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, onMounted, onWillUpdateProps, useRef } from "@odoo/owl";
+import { SpreadsheetChildEnv } from "../../../../types";
+
+interface Props {
+  open: boolean;
+  slots: any;
+}
+
+export class Collapse extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-Collapse";
+  static props = {
+    open: Boolean,
+    slots: Object,
+  };
+
+  private contentRef = useRef("content");
+
+  setup() {
+    onMounted(() => {
+      this.setCollapsedState(this.props.open, true);
+    });
+
+    onWillUpdateProps((nextProps) => {
+      if (nextProps.open !== this.props.open) {
+        this.setCollapsedState(nextProps.open);
+      }
+    });
+  }
+
+  setCollapsedState(open: boolean, skipTransition = false) {
+    const el = this.contentRef.el;
+    if (!el) return;
+
+    if (open) {
+      el.style.maxHeight = el.scrollHeight + "px";
+      el.style.overflow = "hidden";
+      el.style.transition = "max-height 0.3s ease";
+    } else {
+      el.style.maxHeight = "0px";
+      el.style.overflow = "hidden";
+      el.style.transition = "max-height 0.3s ease";
+    }
+    if (skipTransition) {
+      el.style.transition = "";
+    }
```

Comment:
I don't think this is useful ? Deleting it doesn't change anything that I can see. On mounted we go from a state of no max-height to a state with a max-height, there's no css transition to be run there.